src/frontend/flight/handler.rs

```diff
@@ -89,4 +104,121 @@ impl SeafowlFlightHandler {
+                .context
+                .metastore
+                .object_stores
+                .get_log_store_for_table(
+                    Url::parse(&store_loc.location).map_err(|e| {
+                        DataFusionError::Execution(format!(
+                            "Couldn't parse sync location: {e}"
+                        ))
+                    })?,
+                    store_loc.options,
+                    cmd.path,
+                )?,
+        };
+
+        // Check if a table exists yet in the provided location, and if not create one.
+        // If it does fetch its durable sequence number.
+        let dur_seq = if !log_store.is_delta_table_location().await? {
+            // Get the actual table schema by removing the `SEAFOWL_SYNC_DATA_UD_FLAG` column
+            let schema = batches.first().unwrap().schema();
+            let idxs = (0..schema.all_fields().len() - 1).collect::<Vec<usize>>();
+            let schema = schema.project(&idxs)?;
+
+            let delta_schema = DeltaSchema::try_from(&schema)?;
+
+            debug!(
+                "Creating new Delta table at location: {}",
+                log_store.root_uri()
+            );
+            CreateBuilder::new()
+                .with_log_store(log_store.clone())
+                .with_columns(delta_schema.fields().clone())
+                .with_comment(format!("Created by Seafowl {}", env!("CARGO_PKG_VERSION")))
+                .await?;
+            None
+        } else {
+            let mut table = DeltaTable::new(log_store.clone(), Default::default());
+            table.load().await?;
+
+            // TODO: handle all edge cases with missing/un-parsable sequence numbers
+            let commit_infos = table.history(Some(1)).await?;
+            match commit_infos
+                .last()
+                .expect("Table has non-zero commits")
+                .info
+                .get(SEAFOWL_SYNC_DATA_SEQUENCE_NUMBER)
+            {
+                Some(Value::Number(seq)) => seq.as_u64(),
+                _ => None,
+            }
+        };
+
+        // Get the current memory sequence number, defaulting to durable sequence number if missing
+        let url = log_store.root_uri();
+        let mem_seq = self
+            .sync_manager
+            .read()
+            .await
+            .mem_seq_for_table(&url)
+            .or(dur_seq);
+
+        let num_rows = batches
+            .iter()
+            .fold(0, |rows, batch| rows + batch.num_rows());
+        if num_rows == 0 {
+            debug!("Received empty batches, returning current sequence numbers");
+            return Ok(DataSyncResult {
+                accepted: true,
+                memory_sequence_number: mem_seq,
+                durable_sequence_number: dur_seq,
+            });
+        }
+
+        debug!("Processing data change with {num_rows} rows for url {url}");
+        // TODO: make timeout configurable
+        let accepted =
```

Comment:
Come to think about it, we might also need to reject the write if it'll put us above some "hard" memory limit. Maybe we should have a "low watermark" (flush when we're using more RAM than this) and "high watermark" (reject writes to never go above this limit) memory limit. We'll find out after load testing I guess.

Reply:
Maybe? I guess if we're over the low watermark we're going to be flushing syncs into storage and reducing the memory usage anyway, it's not like we're going to continue accumulating stuff in memory.

Reply:
True, if the flushing happens in the same process as the Flight handler, if it happens in the background, we might keep accumulating data even though we started the flush? Guess we'll find out with load testing